test/unit/config/index.js

```diff
@@ -0,0 +1,291 @@
+            });
+
+            it('should throw if `apiToken` is not provided in config', () => {
+                stubConfig_({
+                    path: './docpub.config',
+                    config: {
+                        username: 'random_username',
+                        url: 'random_url'
+                    }
+                });
+
+                expect(() => new Config('./docpub.config')).to.throw(/apiToken/);
+            });
+
+            it('should throw if `apiToken` is not string', () => {
+                stubConfig_({
+                    path: './docpub.config',
+                    config: {
+                        username: 'random_username',
+                        apiToken: 123,
+                        url: 'random_url'
+                    }
+                });
+
+                expect(() => new Config('./docpub.config')).to.throw(/string/);
+            });
+
+            it('should throw if `url` is not provided in config', () => {
+                stubConfig_({
+                    path: './docpub.config',
+                    config: {
+                        username: 'random_username',
+                        apiToken: 'random_token'
+                    }
+                });
+
+                expect(() => new Config('./docpub.config')).to.throw(/url/);
+            });
+
+            it('should throw if `url` is not string', () => {
+                stubConfig_({
+                    path: './docpub.config',
+                    config: {
+                        username: 'random_username',
+                        apiToken: 'random_token',
+                        url: 123
+                    }
+                });
+
+                expect(() => new Config('./docpub.config')).to.throw(/string/);
+            });
+        });
+
+        describe('overrides', () => {
+            let oldArgv;
+            let oldEnv;
+
+            beforeEach(() => {
+                oldArgv = [].concat(process.argv);
+                oldEnv = _.clone(process.env);
+            });
+
+            afterEach(() => {
+                process.env = oldEnv;
+                process.argv = oldArgv;
+            });
+
+            it('should override option in config with CLI argument beginning with `--`', () => {
+                process.argv = process.argv.concat('--username', 'argv_username');
+                stubConfig_({
+                    path: './docpub.config',
+                    config: {
+                        username: 'config_username',
+                        apiToken: 'default_token',
+                        url: 'default_url'
+                    }
+                });
+
+                const config = new Config('./docpub.config');
+
+                expect(config.username).to.be.equal('argv_username');
+            });
+
+            it('should override option in config with ENV variable prefixed with `docpub_`', () => {
```

Comment:
I just noticed your comment about this. I guess ENV having higher priority is okay, as long as it's documented in the README.